api/agent/protocol/json.go

```diff
@@ -141,5 +143,16 @@ func (h *JSONProtocol) Dispatch(ctx context.Context, ci CallInfo, w io.Writer) e
 	}
 
 	_, err = io.WriteString(rw, jout.Body)
+	return h.isExcessData(err, decoder)
+}
+
+func (h *JSONProtocol) isExcessData(err error, decoder *json.Decoder) error {
+	if err == nil {
+		// Now check for excess output, if this is the case, we can be certain that the next request will fail.
+		tmp, ok := decoder.Buffered().(*bytes.Reader)
+		if ok && tmp.Len() > 0 {
+			return ErrExcessData
+		}
+	}
```

Comment:
i guess this has been consumed from the stdout pipe but yet chewed off in the decoder? blimey. it would be nice to do this for all protos (no need here, the mystical 'one day')

Reply:
tried similar things on http, but with no luck... yet.. TODO... 

I'm not sure if this is 100% reliable since it's based on that buf, but hey, if we can catch it, it's a gain...

Reply:
Seems like buffered unread data is lost if we ignore it. Very likely to corrupt next request...